appsec/src/extension/commands_helpers.c

```diff
@@ -648,6 +678,103 @@ bool dd_command_process_metrics(mpack_node_t root, zend_object *nonnull span)
+        }
+
+        size_t key_len = mpack_node_strlen(key);
+        mpack_node_t arr_value = mpack_node_map_value_at(metrics, i);
+
+        for (size_t j = 0; j < mpack_node_array_length(arr_value); j++) {
+            mpack_node_t value = mpack_node_array_at(arr_value, j);
+            mpack_node_t dval_node = mpack_node_array_at(value, 0);
+            double dval = mpack_node_double(dval_node);
+
+            const char *tags_str = "";
+            size_t tags_len = 0;
+            if (mpack_node_array_length(value) >= 2) {
+                mpack_node_t tags = mpack_node_array_at(value, 1);
+                tags_str = mpack_node_str(tags);
+                tags_len = mpack_node_strlen(tags);
+            }
+            if (mpack_node_error(metrics) != mpack_ok) {
+                break;
+            }
+
+            _handle_telemetry_metric(
+                key_str, key_len, dval, tags_str, tags_len);
+        }
+    }
+
+    return true;
+}
+
+static void _init_zstr(
+    zend_string *_Atomic *nonnull zstr, const char *nonnull str, size_t len)
+{
+    zend_string *zstr_cur = atomic_load_explicit(zstr, memory_order_acquire);
+    if (zstr_cur != NULL) {
+        return;
+    }
+    zend_string *zstr_new = zend_string_init(str, len, 1);
+    if (atomic_compare_exchange_strong_explicit(zstr, &(zend_string *){NULL},
+            zstr_new, memory_order_release, memory_order_relaxed)) {
+        return;
+    }
+    zend_string_release(zstr_new);
+}
+
+// NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+void _handle_telemetry_metric(const char *nonnull key_str, size_t key_len,
+    double value, const char *nonnull tags_str, size_t tags_len)
+{
+#define HANDLE_METRIC(name, type)                                              \
+    do {                                                                       \
+        if (key_len == LSTRLEN(name) && memcmp(key_str, name, key_len) == 0) { \
+            static zend_string *_Atomic key_zstr;                              \
+            _init_zstr(&key_zstr, name, LSTRLEN(name));                        \
+            zend_string *tags_zstr = zend_string_init(tags_str, tags_len, 1);  \
+            ddtrace_metric_register_buffer(                                    \
+                key_zstr, type, DDTRACE_METRIC_NAMESPACE_APPSEC);              \
+            ddtrace_metric_add_point(key_zstr, value, tags_zstr);              \
+            zend_string_release(tags_zstr);                                    \
+            mlog_g(dd_log_debug,                                               \
+                "Telemetry metric %.*s added with tags %.*s and value %f",     \
+                (int)key_len, key_str, (int)tags_len, tags_str, value);        \
+            return;                                                            \
+        }                                                                      \
+    } while (0)
+
+    HANDLE_METRIC("waf.requests", DDTRACE_METRIC_TYPE_COUNT);
```

Comment:
I saw you added some integration tests checking this metrics but I am missing some PHPT checking the extension.

Reply:
It would also allow you to test unknown telemetry metrics